vscode-trace-extension/src/utils/backend-tsp-client-provider.ts

```diff
@@ -41,7 +41,23 @@ export const addTspClientChangeListener = (listenerFunction: (tspClient: TspClie
 export async function isUp(): Promise<boolean> {
     const health = await getTspClient().checkHealth();
     const status = health.getModel()?.status;
-    return health.isOk() && status === 'UP';
+    const serverIsUp = health.isOk() && status === 'UP';
+    vscode.commands.executeCommand('setContext', 'traceViewer.serverOn', serverIsUp);
+    return serverIsUp;
+}
+
+/**
+ * Checks for opened traces and updates the vscode context `trace-explorer.noExperiments`
+ * @returns noExperiments as boolean
+ */
+export async function areExperimentsOpened(): Promise<boolean> {
```

Comment:
ATM, the returned boolean is not used. I am a bit disturbed by the reversal in meaning of the returned value vs function name. One potential way forward would be for this function to focus on updating the `noExperiment` context and rename it in consequence. e.g. `updateNoExperimentsContext()`

Reply:
Yea this makes sense.  Addressed.